freqtrade/strategy/interface.py

```diff
@@ -476,8 +481,20 @@ def get_signal(self, pair: str, timeframe: str, dataframe: DataFrame) -> Tuple[b
         (buy, sell) = latest[SignalType.BUY.value] == 1, latest[SignalType.SELL.value] == 1
         logger.debug('trigger: %s (pair=%s) buy=%s sell=%s',
                      latest['date'], pair, str(buy), str(sell))
+        if self.ignore_expired_candle(dataframe=dataframe, buy=buy):
+            return False, sell
         return buy, sell
 
+    def ignore_expired_candle(self, dataframe: DataFrame, buy: bool):
+        if self.ignore_buying_expired_candle and buy:
+            current_time = datetime.now(timezone.utc) - timedelta(
+                seconds=self.ignore_buying_expired_candle_after)
+            candle_time = dataframe['date'].tail(1).iat[0]
+            time_delta = current_time - candle_time
+            return time_delta.total_seconds() > self.ignore_buying_expired_candle_after
```

Comment:
I don't think this is correct / will work as you expect it to

Let's assume it's now 10:01:05 (and i use 1h candles).
* `current_time` would be set to 10:00:05 (it's Now - 60s).
* `candle_time` will be 09:00:00 (it's the candle's open, and the 09:00-10:00 candle is the last closed candle)
* timedelta will therefore be 1h5s - which will always be greater than "ignore_buying_after" - unless i set it to "candle-length + timeout".


While it may sound idiotic at first, consider passing "now" as parameter (assigned in the code to  `datetime.now(timezone.utc)`).

This will simplify testing. Alternatively we could patch `datetime.now()` - but that's usually a dirty hack.

---

on top of that - i'd not use `dataframe['date'].tail(1).iat[0]` ... you're assuming that date is the first column - which is USUALLY correct - but the strategy can change the order at will - so i'd use `.loc[:, 'date']` or `['date']` to get the correct column.

Reply:
thanks for pointing that out, I wasn't aware that the 'date' on the candle was actually the time it started. Some other things I had trouble with became clearer now as well, thanks!

I adjusted the code based on your comments, I think it's better now